net/table_net_certificate.go

```diff
@@ -1,51 +1,79 @@
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
+	"io/ioutil"
 	"net"
+	"net/http"
 	"time"
 
-	"github.com/genkiroid/cert"
+	"golang.org/x/crypto/ocsp"
 
 	"github.com/turbot/steampipe-plugin-sdk/v3/grpc/proto"
 	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
 	"github.com/turbot/steampipe-plugin-sdk/v3/plugin/transform"
 )
 
+type OCSP struct {
+	StatusString           string     `json:"status"`
+	RevokedAt              *time.Time `json:"revoked_at,omitempty"`
+	RevocationReasonString string     `json:"revocation_reason,omitempty"`
+}
+
+//// TABLE DEFINITION
+
 func tableNetCertificate(ctx context.Context) *plugin.Table {
 	return &plugin.Table{
 		Name:        "net_certificate",
 		Description: "Certificate details for a domain.",
 		List: &plugin.ListConfig{
-			Hydrate:    tableNetCertificateList,
-			KeyColumns: plugin.SingleColumn("domain"),
+			Hydrate: tableNetCertificateList,
+			KeyColumns: plugin.KeyColumnSlice{
+				{Name: "domain", Require: plugin.Required, Operators: []string{"="}},
+			},
 		},
 		Columns: []*plugin.Column{
 			// Top columns
 			{Name: "domain", Type: proto.ColumnType_STRING, Description: "Domain name the certificate represents."},
 			{Name: "common_name", Type: proto.ColumnType_STRING, Description: "Common name for the certificate."},
 			{Name: "not_after", Type: proto.ColumnType_TIMESTAMP, Description: "Time when the certificate expires. Also see not_before."},
+			{Name: "is_revoked", Type: proto.ColumnType_BOOL, Hydrate: getRevocationInformation, Description: "Indicates whether the certificate was revoked, or not."},
+			{Name: "transparent", Type: proto.ColumnType_BOOL, Hydrate: getCertificateTransparencyLogs, Transform: transform.FromValue(), Description: "Indicates whether certificate is visible in certificate transparency logs."},
+			{Name: "is_ca", Type: proto.ColumnType_BOOL, Transform: transform.FromField("IsCertificateAuthority"), Description: "True if the certificate represents a certificate authority."},
 			// Other columns
+			{Name: "serial_number", Type: proto.ColumnType_STRING, Description: "Serial number of the certificate."},
+			{Name: "subject", Type: proto.ColumnType_STRING, Description: "Subject of the certificate."},
+			{Name: "public_key_algorithm", Type: proto.ColumnType_STRING, Description: "Public key algorithm used by the certificate."},
+			{Name: "public_key_length", Type: proto.ColumnType_INT, Description: "Specifies the size of the key."},
+			{Name: "signature_algorithm", Type: proto.ColumnType_STRING, Description: "Signature algorithm of the certificate."},
+			{Name: "ip_address", Type: proto.ColumnType_IPADDR, Transform: transform.FromField("IPAddress"), Description: "IP address associated with the domain."},
+			{Name: "issuer", Type: proto.ColumnType_STRING, Description: "Issuer of the certificate."},
+			{Name: "issuer_name", Type: proto.ColumnType_STRING, Description: "Issuer of the certificate."},
```

Comment:
This column and `issuer` have the same description, are there better descriptions we can use to better differentiate them?